File: Dr.NutrizioNino.Models/Dto/DishDetailDto.cs
namespace Dr.NutrizioNino.Models.Dto;

public record DishDetailNutrientDto(
    Guid NutrientId,
    string Name,
    int PositionOrder,
    Guid UnitOfMeasureId,
    decimal Quantity
);

public record DishDetailIngredientDto(
    Guid FoodId,
    string FoodName,
    decimal QuantityGrams
);

public record DishDetailDto(
    Guid Id,
    string Name,
    decimal WeightGrams,
    IList<DishDetailIngredientDto> Ingredients,
    IList<DishDetailNutrientDto> Nutrients
);

public record RescaleDishRequest(decimal WeightGrams, bool Recalculate = false);

File: src/Dr.NutrizioNino.WebVue/src/Interfaces/dishes/DishDetailDto.ts
export interface DishDetailNutrientDto {
  nutrientId: string
  name: string
  positionOrder: number
  unitOfMeasureId: string
  quantity: number
}

export interface DishDetailIngredientDto {
  foodId: string
  foodName: string
  quantityGrams: number
}

export interface DishDetailDto {
  id: string
  name: string
  weightGrams: number
  ingredients: DishDetailIngredientDto[]
  nutrients: DishDetailNutrientDto[]
}
